samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -65,16 +65,25 @@
   private volatile String ephemeralPath = null;
   private final ZkKeyBuilder keyBuilder;
   private final int connectionTimeoutMs;
+  private ZkJobCoordinatorMetrics metrics;
 
   public ZkUtils(ZkKeyBuilder zkKeyBuilder, ZkClient zkClient, int connectionTimeoutMs) {
     this.keyBuilder = zkKeyBuilder;
     this.connectionTimeoutMs = connectionTimeoutMs;
     this.zkClient = zkClient;
   }
 
+  public ZkUtils(ZkKeyBuilder zkKeyBuilder, ZkClient zkClient, int connectionTimeoutMs, ZkJobCoordinatorMetrics metrics) {
+    this.keyBuilder = zkKeyBuilder;
+    this.connectionTimeoutMs = connectionTimeoutMs;
+    this.zkClient = zkClient;
+    this.metrics = metrics;
```

Comment:
@sborya I already committed this JIRA a few days back. 
@santhoshvenkat1988 identified this issue and is in the process of fixing it. https://builds.apache.org/view/S-Z/view/Samza/job/samza-pr-checks/663/